packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx

```diff
@@ -119,20 +121,55 @@ const DefaultFilterInputs = <T extends ConditionalRule>({
         case FilterOperator.NOT_IN_THE_PAST:
         case FilterOperator.IN_THE_NEXT:
         case FilterOperator.IN_THE_CURRENT:
-        case FilterOperator.IN_BETWEEN:
-            return (
-                <FilterNumberInput
-                    disabled={disabled}
-                    placeholder={placeholder}
-                    value={rule.values?.[0]}
-                    onChange={(newValue) => {
-                        onChange({
-                            ...rule,
-                            values: newValue ? [newValue] : [],
-                        });
-                    }}
-                />
-            );
+        case FilterOperator.IN_BETWEEN: {
+            switch (filterType) {
+                case FilterType.STRING:
```

Comment:
perhaps this (and the case in 115) can be moved to its own `case`? I think it's a bit confusing 🤔

Reply:
Thank you @jogeshwar01 for the prompt push. I will review this locally and with a duplicate PR to test e2es :)